crates/wasi-common/src/old/snapshot_0/fdpool.rs

```diff
@@ -0,0 +1,132 @@
+use std::collections::{HashSet, VecDeque};
```

Comment:
Could this file reexport from the current snapshot to avoid duplication?

Reply:
Excellent question! I thought about this myself, and in principle, I'd love if we could. The problem is with type aliasing between the two snapshots. The compiler gets confused which `wasi::__wasi_fd_t as Fd` to use. Do you think there's a way to overcome this? Would perhaps type aliasing work here?

Reply:
Hm I'm not sure I understand the compiler error, wanna push up a branch I can poke at?

Reply:
Right, I now remembered what the issue was. Since `__wasi_fd_t` is a type alias to `u32` in both snapshots, we obviously cannot define two implementations of the form

```rust
// this one is in snapshot1, in wasi.rs
impl crate::fdpool::Fd for __wasi_fd_t {
   // ...
}
```

and

```rust
// this one is in snapshot0, in old/snapshot0/wasi.rs
impl crate::fdpool::Fd for __wasi_fd_t {
    // ...
}
```

I guess given that there is virtually no ABI difference between the two types across both snapshots, we could simply ignore implementing it in snapshot0. Would that be acceptable? I mean given that soon-ish we'd want to use `wiggle` to completely rethink how we handle multiple snapshots with as little code duplication as possible, this seems fair to me.

Reply:
Yeah I think that'll be alright!